ssz_serialization/merkleization.nim

```diff
@@ -604,16 +607,13 @@ func hashTreeRootAux[T](x: T): Digest =
     unsupported T
 
 func hashTreeRootList(x: List|BitList): Digest =
-  const maxLen = static(x.maxLen)
-  type T = type(x)
-  const limit = maxChunksCount(T, maxLen)
-  var merkleizer = createMerkleizer(limit)
-
   when x is BitList:
-    let contentsHash = bitListHashTreeRoot(merkleizer, BitSeq x)
+    const totalChunks = maxChunksCount(T, x.maxLen)
```

Comment:
I'm confused, where does the `T` come from?

Reply:
Oh, never mind, it gets then moved in this commit: https://github.com/status-im/nim-ssz-serialization/pull/9/commits/722502dcb810f02a526822d2da750b82c6480242

Reply:
Thanks. I fixed it so that the intermediate commit now also is correct!